openssl/src/x509/mod.rs

```diff
@@ -64,6 +64,34 @@ foreign_type! {
     pub struct X509StoreContextRef;
 }
 
+pub struct X509StoreContextBuilder(X509StoreContext);
+
+impl X509StoreContext {
+    pub fn builder() -> Result<X509StoreContextBuilder, ErrorStack> {
+        unsafe {
+            ffi::init();
+            cvt_p(ffi::X509_STORE_CTX_new()).map(|p| X509StoreContextBuilder(X509StoreContext(p)))
+        }
+    }
+
+    #[cfg(any(all(feature = "v102", ossl102), all(feature = "v110", ossl110)))]
+    pub fn verify_cert(self) -> Result<(), ErrorStack> {
```

Comment:
Actually, I intentionally chose `self` based on this from the OpenSSL docs:

> A negative return value from X509_verify_cert() can occur if it is invoked incorrectly, such as with no certificate set in ctx, or when it is called twice in succession without reinitialising ctx for the second call.

https://www.openssl.org/docs/man1.0.2/crypto/X509_verify_cert.html

Basically, the X509StoreContext needs to be reinitialized (which I do in the `build()` method). If you'd prefer some other mechanism to make a similar guarantee, let me know.